pkg/scm/git/git.go

```diff
@@ -28,18 +31,50 @@ type Git interface {
 	Clone(source, target string, opts api.CloneConfig) error
 	Checkout(repo, ref string) error
 	SubmoduleUpdate(repo string, init, recursive bool) error
+	LsTree(repo, ref string, recursive bool) ([]os.FileInfo, error)
 	GetInfo(string) *api.SourceInfo
+	CreateEmptyLocalGitDirectory() (string, error)
+	CreateLocalGitDirectory() (string, error)
+	CreateLocalGitDirectoryWithSubmodule() (string, error)
+}
+
+func isUsingCygwinGit() bool {
+	if runtime.GOOS == "windows" {
+		// If we find the cygpath utility (which translates between UNIX-style and
+		// Windows-style paths) in the same directory as git, assume we're using
+		// Cygwin.
+		cygpath, _ := exec.LookPath("cygpath")
```

Comment:
@jim-minter, just making sure I understand what you meant here... we ignore the error because we rely on the `cygpath` variable being the empty string in case of errors and then it would fail the comparison with the `git` path later? Is that how you reasoned here?